plugin/settings/settings_storage.py

```diff
@@ -0,0 +1,209 @@
+                log.debug("  %-26s <-- '%s'", setting_name, val)
+        log.debug(" Settings sucessfully read...")
+
+        # initialize max_tu_age if is it not yet, default to 30 minutes
+        self.max_tu_age = getattr(self, "max_tu_age", "00:30:00")
+        # get seconds from string if needed
+        if isinstance(self.max_tu_age, str):
+            self.max_tu_age = Tools.seconds_from_string(self.max_tu_age)
+
+    def __populate_common_flags(self, view):
+        """ Populate the variables inside common_flags with real values
+
+        Args:
+            view (sublime.View): current view
+
+        Returns:
+            str[]: clang common flags with variables expanded
+
+        """
+        # init current and parrent folders:
+        if not view.file_name():
+            log.error(" no view to populate common flags from")
+            return
+        file_current_folder = path.dirname(view.file_name())
+        file_parent_folder = path.dirname(file_current_folder)
+
+        # init wildcard variables
+        self.__update_widcard_values()
+
+        # populate variables to real values
+        log.debug(" populating common_flags with current variables.")
+        for idx, flag in enumerate(self.common_flags):
+            self.common_flags[idx] = self.__replace_wildcard_if_needed(flag)
+
+        if self.include_file_folder:
+            self.common_flags.append("-I" + file_current_folder)
+        if self.include_file_parent_folder:
+            self.common_flags.append("-I" + file_parent_folder)
+
+    def __replace_wildcard_if_needed(self, flag):
+        """ Replace wildcards in a flag if they are present there
+
+        Args:
+            flag (str): flag possibly with wildcards in it
+
+        Returns:
+            str: flag with replaced wildcards
+        """
+        for wildcard, value in self._wildcard_values.items():
+            res = re.sub(re.escape(wildcard), value, flag)
+            if res != flag:
+                log.debug(" populating '%s': '%s'", wildcard, res)
+                return res
+        return flag
+
+    def __update_widcard_values(self):
+        """ Update values for wildcard variables
+        """
+        variables = sublime.active_window().extract_variables()
+        if 'folder' in variables:
+            project_folder = variables['folder'].replace('\\', '\\\\')
+            self._wildcard_values[Wildcards.PROJECT_PATH] = project_folder
+        if 'project_base_name' in variables:
+            project_name = variables['project_base_name'].replace('\\', '\\\\')
+            self._wildcard_values[Wildcards.PROJECT_NAME] = project_name
+
+        # duplicate as fields
+        self.project_base_folder = self._wildcard_values[Wildcards.PROJECT_PATH]
+        self.project_base_name = self._wildcard_values[Wildcards.PROJECT_NAME]
+
+        # get clang version string
+        self._wildcard_values[Wildcards.CLANG_VERSION] =\
+            Tools.get_clang_version_str(self.clang_binary)
```

Comment:
This uses potentially view-overridden clang_binary path. This seems a bit wrong as completers are created based on clang_binary from user settings (not view specific).
I hope we can fix it so that per-view clang_binary is always respected, but that might require bigger changes.